### ISSUE TITLE
`rust.unknown_lint = "allow"` not honored when unknown lint name is similar to existing lint

### ISSUE THREAD

**Issue by danielrainer** [rust-lang/rust#151179]
### Summary

When `rust.unknown_lints = "allow"` is set, warnings for unknown lints are sometimes shown, seemingly when their name is close enough to an existing lint.

### Lint Name

`rust.unknown_lints = "allow"`

### Reproducer

Create a new default Rust project using `cargo init` and put the following lint section into `Cargo.toml`:
```toml
[lints]
rust.unknown_lints = "allow"
# silently ignored
clippy.un = "warn"
# warning about unknown lint
clippy.unn = "warn"
```
Then run `cargo clippy`.

I saw this happen:

```text
warning[E0602]: unknown lint: `clippy::unn`
  |
  = help: did you mean: `clippy::all`
  = note: requested on the command line with `-W clippy::unn`
  = note: `#[warn(unknown_lints)]` on by default
```

I expected to see this happen:

No warnings should be shown, since `rust.unknown_lints = "allow"` is set. But even if that is ignored, there should be a warning about both `clippy.un` and `clippy.unn`, which is what happens if the `rust.unknown_lints` setting is removed.

It seems that checking for similarly-named lints is enabled when `rust.unknown_lints = "allow"` is set, when instead it should be disabled in this situation. This behavior is present in nightly and goes back to at least 1.85.

### Version

```text
rustc 1.94.0-nightly (86a49fd71 2026-01-14)
binary: rustc
commit-hash: 86a49fd71fecd25b0fd20247db0ba95eeceaba28
commit-date: 2026-01-14
host: x86_64-unknown-linux-gnu
release: 1.94.0-nightly
LLVM version: 21.1.8
```
